packages/server/routes/api/auth.js

```diff
@@ -107,4 +114,88 @@ router.get("/google/callback", passport.authenticate("google", {
   response.redirect(`icebreak://login?token=${newToken}`);
 });
 
-module.exports = router;
\ No newline at end of file
+router.post("/register", async (request, response) => {
+
+  try{
+    const { email, password, } = request.body;
+    
+    let emailCheckRegex = /^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/g;
+
+    if(!emailCheckRegex.test(email)){ // check if email is valid, doesn't include or no spaces
+      throw new Error("Email is invalid.");
+    }
+
+    const ifExist = await user.getUserByEmail(email);
+
+    if(ifExist?.email === email){ // check if email is already in the database
+      throw new Error("User already exists with this email.");
+    }
+
+    const saltRounds = 10;
+
+    bcrypt.genSalt(saltRounds, function(err, salt) {
+      bcrypt.hash(password, salt, async function(err, hash) { // encrypt the user password using bcrypt
+
+        // create unique User ID as bytes (18 byte)
+        const user_id = uniqid();
+
+        await postgres.query(`
+          INSERT INTO users (user_id, first_name, last_name, email, avatar, password)
+          VALUES ('${user_id}', 'firstName', 'lastName', '${email}', 'avatar', '${password}');
+        `); // create new user in DB
+
+        const newToken = token.generate({ email, hash}); // create a signed jwt token
```

Comment:
Do not store sensitive data inside the JWT token, it's not a encryption token.

The payload being passed into the token is in plain text and can be seen by anyone.
https://jwt.io/